src/DotNetEnv/Parsers.cs

```diff
@@ -176,14 +179,17 @@ private static string ToUtf32Char (string hex)
         // unquoted values can have interpolated variables,
         // but only inline whitespace -- until a comment,
         // and no escaped chars, nor byte code chars
-        // FIXME: would be nice to solve multi word with comment case with parser directly (no split + trim)
         internal static readonly Parser<ValueCalculator> UnquotedValue =
             InterpolatedValue.Or(
-                NotControlNorWhitespace("'\"$")
-                .Or(InlineWhitespaceChars.AtLeastOnce().Text())
-                .AtLeastOnce()
-                .Select(strs => new ValueActual(strs))
-            ).Many().Select(vs => new ValueCalculator(vs).Split("[ \t]#").Trim());
+                InlineWhitespaceChars.Many().Then(w =>
+                    NotControlNorWhitespaceChar("#'\"$").Once().Then(
```

Comment:
This includes a behavior change! Per https://github.com/tonerdo/dotnet-env/pull/82#discussion_r1424412109 and https://github.com/tonerdo/dotnet-env/pull/82#issuecomment-1854305046

Reply:
Not sure about the leading `InlineWhitespaceChars` here. Seems like unquoted values are allowed to have leading whitespaces now, but thats definitely unintended in my opinion.
I would expect unquoted values to never begin or end with a whitespace. (to be discussed: what happens on Interpolation? I would say with Interpolation leading/trailing whitespaces are allowed for unquoted values, and removing the Trim is a good step towards this goal --> we should add a test for this, either way ;) )

The way with Except was much better to read in my opinion, but as we have that point wheter my version drops something or not I will explore a little more, adapt your new tests and give a more elaborated response.

Reply:
The logic here is that an unquoted is zero or more of two sub patterns concatenated: an interpolated value, or:

Zero or more whitespace chars
Followed by exactly one non special char (i.e. no comments and no quotes, and no dollar sign like normal, for interpolated vars, and non control or whitespace, to not conflict with the whitespace pattern) to start a word
Followed by zero or more non dollar sign chars (and non control or whitespace again)
And then all of that makes a single "value" to go along with interpolated vars.